src/structures/Guild.js

```diff
@@ -392,19 +392,23 @@ class Guild extends Base {
    */
 
   /**
-   * Fetches a collection of banned users in this guild.
-   * @returns {Promise<Collection<Snowflake, BanInfo>>}
-   */
-  fetchBans() {
-    return this.client.api.guilds(this.id).bans.get().then(bans =>
-      bans.reduce((collection, ban) => {
+   * Fetches a single banned user or a collection of banned users in this guild.
+   * @param {string} [id] Optional id of a banned user.
+   * @returns {Promise<Collection<Snowflake, BanInfo>> | Promise<BanInfo>}
+   */
+  fetchBans(id) {
+    return this.client.api.guilds(this.id).bans.get(id).then(data => {
```

Comment:
`id` is misplaced. That should be:
`return this.client.api.guilds(this.id).bans(id).get().then(data => {`